pkg/libdevfile/libdevfile.go

```diff
@@ -119,3 +119,55 @@ func execDevfileEvent(devfileObj parser.DevfileObj, events []string, handler Han
 	}
 	return nil
 }
+
+// GetContainerComponents returns a slice of container components in the given devfile
+func GetContainerComponents(devfileObj parser.DevfileObj) ([]v1alpha2.Component, error) {
+	containers, err := devfileObj.Data.GetComponents(common.DevfileOptions{
+		ComponentOptions: common.ComponentOptions{ComponentType: v1alpha2.ContainerComponentType},
+	})
+	if err != nil {
+		return nil, err
+	}
+	return containers, nil
+}
+
+// GetContainerEndpointMapping returns a map of container names and slice of its endpoints (in int) with exposure status other than none
+func GetContainerEndpointMapping(containers []v1alpha2.Component) (map[string][]int, error) {
+	ceMapping := make(map[string][]int)
+	if len(containers) == 0 {
+		return ceMapping, nil
+	}
+
+	for _, cntr := range containers {
```

Comment:
You can use a more readable name, like `container`

Hopefully, we are not programming in asembler anymore ;)

Reply:
Haha. I just wanted to have a shorter name for that temporary variable. Changing it.

Reply:
I added the unit test along with fixing it.